Formula/cadence-workflow.rb

```diff
@@ -0,0 +1,33 @@
+class CadenceWorkflow < Formula
+  desc "Cadence Workflow is a distributed, scalable, durable, and highly available orchestration engine to execute asynchronous long-running business logic in a scalable and resilient way"
+  homepage "https://cadenceworkflow.io/"
+  url "https://github.com/uber/cadence.git",
+      tag:      "v0.18.1",
+      revision: "b6dfe37ebfec7742b1470be7129f07518e8bec98"
+  license "MIT"
+
+  bottle do
+  end
+
+  depends_on "go" => :build
+
+  conflicts_with "cadence", because: "both install an `cadence` executable"
+
+  def install
+    ENV["PATH"] = "#{ENV["PATH"]}:#{ENV["GOPATH"]}/bin"
```

Comment:
Why is this needed?

Reply:
@git-hulk can you help answer it? 
I guess it's because the binary requires libraries from GOPATH? is that correct?

Reply:
`ENV["GOPATH"]` should return `nil`, so I believe this just appends `/bin` to `PATH`, but `/bin` should already be in `PATH`.

Reply:
When I search `grep GOPATH * -A 3 -B 1` in formulas, there are many different ways of doing it. 

For example, this
```
  def install
    ENV["GOPATH"] = buildpath
    ENV["GO111MODULE"] = "auto"
    ENV.prepend_create_path "PATH", buildpath/"bin"
```
does't work for us.

Reply:
Yea, those existing formulae shouldn't be using `GOPATH` anymore. `GOPATH` usage is deprecated in Go (modules should be used instead), and we're no longer accepting formulae that need it in homebrew/core.

However, it looks like building using Go modules should work for this formula.

Reply:
Yes, there is a script executed in Makefile to compose LDFLAG: https://github.com/uber/cadence/blob/7bd91051360214181464b87325615631a1712bb7/Makefile#L71

It's probably too complicated. Is there any concern of doing in a way of this PR?

Reply:
>ENV["GOPATH"] should return nil, so I believe this just appends /bin to PATH, but /bin should already be in PATH.
Interesting, in my laptop(mac) with latest go(1.16.3), do still have that GOPATH:

```
(cadence)$go version
go version go1.16.3 darwin/amd64

(cadence)$echo $GOPATH
/Users/qlong/go

```

Reply:
Actually, my laptop env doesn't need that line. Because PATH already has it.
So I think we could remove it.

Reply:
@carlocab I removed it. LMK if you have any other suggestion. Thanks again :D